File: frontend/src/types/approval.ts
/**
 * 审批类型定义
 * 包含审批流程、审批记录、审批历史等相关类型
 */

/**
 * 审批类型
 */
export type ApprovalType = 'enterprise' | 'member' | 'asset_submit';

/**
 * 审批子类型
 */
export type ApprovalSubType = 'create' | 'update' | 'join';

/**
 * 审批状态
 */
export type ApprovalStatus = 'pending' | 'approved' | 'rejected' | 'returned';

/**
 * 目标类型
 */
export type TargetType = 'enterprise' | 'member';

/**
 * 审批操作类型
 */
export type ApprovalAction = 'approve' | 'reject' | 'return';

/**
 * 审批优先级
 */
export type ApprovalPriority = 'low' | 'medium' | 'high' | 'urgent';

/**
 * 申请人信息
 */
export interface ApplicantInfo {
  userId: string;
  name: string;
  email: string;
  avatar?: string;
}

/**
 * 附件信息
 */
export interface ApprovalAttachment {
  fileId: string;
  fileName: string;
  fileUrl: string;
  fileSize?: number;
  fileType?: string;
}

/**
 * 变更信息
 */
export interface ChangeInfo {
  old: string;
  new: string;
}

/**
 * 目标信息
 */
export interface TargetInfo {
  enterpriseId: string;
  enterpriseName: string;
  changes?: Record<string, ChangeInfo>;
  [key: string]: unknown;
}

/**
 * 操作人信息
 */
export interface OperatorInfo {
  userId: string;
  name: string;
  role: string;
}

/**
 * 审批记录（匹配后端返回的字段）
 */
export interface Approval {
  id: string;
  type: string;
  target_id: string;
  target_type: string;
  applicant_id: string;
  status: string;
  current_step: number;
  total_steps: number;
  remarks: string;
  attachments: ApprovalAttachment[] | null;
  changes: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
  completed_at: string | null;
  asset_id?: string;
  asset_name?: string;
  asset_type?: string;
  enterprise_name?: string;
}

/**
 * 审批流程记录
 */
export interface ApprovalProcessRecord {
  id: string;
  step: number;
  action: string;
  operator_id: string;
  operator_role: string;
  comment: string;
  attachments: ApprovalAttachment[] | null;
  created_at: string;
}

/**
 * 审批详情
 */
export interface ApprovalDetail extends Approval {
  processHistory: ApprovalProcessRecord[];
}

/**
 * 审批列表请求参数
 */
export interface ApprovalListParams {
  page?: number;
  pageSize?: number;
  type?: ApprovalType;
  status?: ApprovalStatus;
  subType?: ApprovalSubType;
  startDate?: string;
  endDate?: string;
  keyword?: string;
  priority?: ApprovalPriority;
}

/**
 * 审批列表响应
 */
export interface ApprovalListResponse {
  total: number;
  page: number;
  pageSize: number;
  items: Approval[];
}

/**
 * 审批操作请求
 */
export interface ApprovalActionRequest {
  action: ApprovalAction;
  comment: string;
  attachments?: ApprovalAttachment[];
}

/**
 * 审批操作响应
 */
export interface ApprovalActionResponse {
  id: string;
  type: string;
  status: string;
}

/**
 * 审批统计
 */
export interface ApprovalStatistics {
  total: number;
  pending: number;
  approved: number;
  rejected: number;
  returned: number;
  todayPending: number;
  urgentPending: number;
}

File: backend/app/schemas/approval.py
"""审批相关数据模型。"""
from datetime import datetime
from typing import Any, Dict, List, Optional
from uuid import UUID

from pydantic import BaseModel, Field, ConfigDict


class AttachmentRequest(BaseModel):
    """附件请求模型。"""
    
    file_name: str = Field(..., description="文件名")
    file_url: str = Field(..., description="文件URL")
    file_type: Optional[str] = Field(None, description="文件类型")
    file_size: Optional[int] = Field(None, description="文件大小（字节）")


class AttachmentResponse(BaseModel):
    """附件响应模型。"""
    
    model_config = ConfigDict(from_attributes=True)
    
    file_name: str = Field(..., description="文件名")
    file_url: str = Field(..., description="文件URL")
    file_type: Optional[str] = Field(None, description="文件类型")
    file_size: Optional[int] = Field(None, description="文件大小（字节）")


class ApprovalCreateRequest(BaseModel):
    """审批申请请求模型。"""
    
    target_id: UUID = Field(..., description="目标对象ID（企业ID或成员关系ID）")
    target_type: str = Field(..., description="目标类型: enterprise, member")
    type: str = Field(..., description="审批类型: enterprise_create, enterprise_update, member_join")
    remarks: Optional[str] = Field(None, description="申请备注")
    attachments: Optional[List[AttachmentRequest]] = Field(None, description="附件列表")
    changes: Optional[Dict[str, Any]] = Field(None, description="变更内容（JSON格式）")


class ApprovalProcessRequest(BaseModel):
    """审批处理请求模型。"""
    
    action: str = Field(..., description="操作类型: approve, reject, return")
    comment: Optional[str] = Field(None, description="审批意见")
    attachments: Optional[List[AttachmentRequest]] = Field(None, description="附件列表")


class ApprovalResponse(BaseModel):
    """审批响应模型。"""
    
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID = Field(..., description="审批ID")
    type: str = Field(..., description="审批类型")
    target_id: UUID = Field(..., description="目标对象ID")
    target_type: str = Field(..., description="目标类型")
    applicant_id: UUID = Field(..., description="申请人ID")
    status: str = Field(..., description="审批状态")
    current_step: int = Field(..., description="当前步骤")
    total_steps: int = Field(..., description="总步骤数")
    remarks: Optional[str] = Field(None, description="申请备注")
    attachments: Optional[List[Dict[str, Any]]] = Field(None, description="附件列表")
    changes: Optional[Dict[str, Any]] = Field(None, description="变更内容")
    created_at: datetime = Field(..., description="创建时间")
    updated_at: datetime = Field(..., description="更新时间")
    completed_at: Optional[datetime] = Field(None, description="完成时间")


class ApprovalProcessHistoryResponse(BaseModel):
    """审批流程历史响应模型。"""
    
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID = Field(..., description="流程记录ID")
    step: int = Field(..., description="步骤序号")
    action: str = Field(..., description="操作类型")
    operator_id: UUID = Field(..., description="操作人ID")
    operator_role: Optional[str] = Field(None, description="操作人角色")
    comment: Optional[str] = Field(None, description="审批意见")
    attachments: Optional[List[Dict[str, Any]]] = Field(None, description="附件列表")
    created_at: datetime = Field(..., description="操作时间")


class ApprovalDetailResponse(BaseModel):
    """审批详情响应模型。"""
    
    id: UUID = Field(..., description="审批ID")
    type: str = Field(..., description="审批类型")
    target_id: UUID = Field(..., description="目标对象ID")
    target_type: str = Field(..., description="目标类型")
    applicant_id: UUID = Field(..., description="申请人ID")
    status: str = Field(..., description="审批状态")
    current_step: int = Field(..., description="当前步骤")
    total_steps: int = Field(..., description="总步骤数")
    remarks: Optional[str] = Field(None, description="申请备注")
    attachments: Optional[List[Dict[str, Any]]] = Field(None, description="附件列表")
    changes: Optional[Dict[str, Any]] = Field(None, description="变更内容")
    created_at: datetime = Field(..., description="创建时间")
    updated_at: datetime = Field(..., description="更新时间")
    completed_at: Optional[datetime] = Field(None, description="完成时间")
    process_history: List[ApprovalProcessHistoryResponse] = Field(default_factory=list, description="流程历史")


class NotificationResponse(BaseModel):
    """通知响应模型。"""
    
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID = Field(..., description="通知ID")
    type: str = Field(..., description="通知类型")
    recipient_id: UUID = Field(..., description="接收人ID")
    approval_id: Optional[UUID] = Field(None, description="关联审批ID")
    title: str = Field(..., description="通知标题")
    content: Optional[str] = Field(None, description="通知内容")
    is_read: bool = Field(..., description="是否已读")
    read_at: Optional[datetime] = Field(None, description="阅读时间")
    created_at: datetime = Field(..., description="创建时间")


class NotificationListResponse(BaseModel):
    """通知列表响应模型。"""
    
    items: List[NotificationResponse] = Field(default_factory=list, description="通知列表")
    total: int = Field(..., description="总记录数")
    page: int = Field(..., description="当前页码")
    page_size: int = Field(..., description="每页数量")
    total_pages: int = Field(..., description="总页数")
    unread_count: int = Field(default=0, description="未读数量")
